pkg/cli/admin/ocpcertificates/certregen/leavesregen.go

```diff
@@ -0,0 +1,94 @@
+package certregen
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	applycorev1 "k8s.io/client-go/applyconfigurations/core/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/openshift/library-go/pkg/operator/certrotation"
+
+	"github.com/openshift/oc/pkg/cli/admin/ocpcertificates/certgraphanalysis"
+)
+
+type LeavesRegen struct {
+	ValidBefore *time.Time
+}
+
+// split here for convenience of unit testing
+func (o *LeavesRegen) forceRegenerationOnSecret(objPrinter *objectPrinter, kubeClient kubernetes.Interface, secret *corev1.Secret, dryRun bool) error {
+	if len(secret.Annotations[certrotation.CertificateIssuer]) == 0 ||
+		len(secret.Annotations[certrotation.CertificateNotBeforeAnnotation]) == 0 ||
+		isRevisioned(secret.ObjectMeta) {
+		// TODO this should return an error if the name was specified.
+		// otherwise, not for this command.
+		return nil
+	}
+
+	keyPairInfo, err := certgraphanalysis.InspectSecret(secret)
+	if err != nil {
+		return fmt.Errorf("error interpretting content: %w", err)
+	}
+	if keyPairInfo.Spec.Details.SignerDetails != nil {
+		// not for this command.
+		return nil
+	}
+
+	issuerInfo := keyPairInfo.Spec.CertMetadata.CertIdentifier.Issuer
+	if issuerInfo == nil {
+		// what are you??
+		return nil
+	}
+
+	if issuerInfo.CommonName == keyPairInfo.Spec.CertMetadata.CertIdentifier.CommonName {
+		// not for this command, we only want leaf certs
+		return nil
+	}
+	if o.ValidBefore != nil {
+		notBefore, err := time.Parse(time.RFC3339, secret.Annotations[certrotation.CertificateNotBeforeAnnotation])
+		if err != nil {
+			return fmt.Errorf("error parsing notBefore: %w", err)
+		}
+		if notBefore.After(*o.ValidBefore) {
+			// not for us
+			return nil
+		}
+	}
+
+	applyOptions := metav1.ApplyOptions{
+		Force:        true,
+		FieldManager: RegenerateSignersFieldManager,
```

Comment:
This field manager needs to be different.  If you use the same field manager and set different fields on an object, it will clear fields you no longer care about.  While probably ok here, we don't want to risk cross talk.

Reply:
good catch